File: backend/Models/Loan.cs
namespace Backend.Models;
using Backend.DTOs;
using System.Text.Json.Serialization;

public class Loan : BaseModel
{
    public bool Returned { get; set; } = false;
    
    [JsonIgnore]
    public int UserId { get; set; }

    [JsonIgnore]
    public int CopyId { get; set; }
    public Copy Copy { get; set; } = null!;

    [JsonIgnore]
    public User User { get; set; } = null!;
    public DateTime DateLoaned { get; set; }
    public DateTime DateDue { get; set; }
    public UserRegResponseDTO? UserInfo => User != null ? UserRegResponseDTO.FromUser(User) : null;
}

File: frontend/src/types/loan.ts
import { Base } from "./base";
import { Copy } from "./copy";
import { Pagination } from "./pagination";
import { User } from "./user";

export interface Loan extends Base {
    user: User,
    copy: Copy,
    dateLoaned: Date,
    dateDue: Date,
    returned: boolean
}

export type LoanFilter = {
    filter: "Ongoing" | "Expired" | null,
    pagination: Pagination | null,
}

export type CreateLoan = {
    userId: number,
    copyId: number
}

export interface UpdateLoan {
    id: number,
    userId: number,
    copyId: number,
    returned: boolean | null,
    dueDate: Date | null
}

